### PR TITLE
finished chapter 1

### PR DESCRIPTION
**Pull request by keuhdall**
### Solutions for Chapter 1

cc @vrom911 @chshersh

### CODE REVIEW

File: src/Chapter1.hs [hunk 1]
```diff
@@ -417,6 +417,7 @@ task is to specify the type of this function.
 49
 -}
 
+squareSum :: Num a => a -> a -> a
```
Comment (vrom911) [inline src/Chapter1.hs:420]: We explain `Num` in the following chapters, so you are one step ahead 😸

In here `Int -> Int -> Int` would also work, as a more specific type 🙂

File: src/Chapter1.hs [hunk 2]
```diff
@@ -478,7 +479,8 @@ Implement a function that returns the last digit of a given number.
   whether it works for you!
 -}
 -- DON'T FORGET TO SPECIFY THE TYPE IN HERE
-lastDigit n = error "lastDigit: Not implemented!"
+lastDigit :: Int -> Int
+lastDigit n = abs $ n `rem` 10
```
Comment (vrom911) [inline src/Chapter1.hs:483]: Interesting solution 👍🏼

File: src/Chapter1.hs [hunk 3]
```diff
@@ -620,7 +627,10 @@ Try to introduce variables in this task (either with let-in or where) to avoid
 specifying complex expressions.
 -}
 
-sumLast2 n = error "sumLast2: Not implemented!"
+sumLast2 :: Int -> Int
+sumLast2 n = last2 $ abs n where
+  last2 x = last1 x + last1 (x `quot` 10)
+  last1 x = x `rem` 10
```
Comment (vrom911) [inline src/Chapter1.hs:633]: For `last1` you could utilise the `lastDigit` function from the above 😉 

Also note, that to make your solution even shorter you could find the standard [`quotRem`](https://hackage.haskell.org/package/base-4.14.0.0/docs/Prelude.html#v:quotRem) function handy, that actually combines inside both `quot` and `rem` in one go.

So you could write it this way:

```haskell
(first, second) = quoteRem x 10
```

You can see how we could pattern match on the pair 🙂

**Review by vrom911 — APPROVED** [review]
Awesome work! Enjoyed your solutions 👍🏼 
Congrats with finishing this Chapter 🥳 
Feel free to merge this PR as it is, it looks great 🥇

**Comment by keuhdall** [comment]
Thank you for your feedback, forgot about quoteRem which would've been indeed much more elegant in the `sumLast2` exercise ! Thank you for your valuable feedback and for this great initiative ! Hopefully I'll motivate some colleagues to try out Haskell as well 🙂

**Pull request merged**